extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMapSessionInstaller.kt

```diff
@@ -0,0 +1,102 @@
+package com.mapbox.maps.extension.androidauto
+
+import androidx.car.app.Session
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import com.mapbox.maps.MapInitOptions
+import com.mapbox.maps.MapboxExperimental
+
+/**
+ * When attaching the many possible experiences onto [MapboxCarMap], this installer allows you to
+ * remove all the boilerplate. Use this installer for creating experiences that will be
+ * attached as long as the head unit is using your application. To assign an experience to a screen
+ * please use the [MapboxCarMapScreenInstaller].
+ *
+ * Use this to setup the [Session]. Use the [Session.mapboxMapInstaller] extension!
+ *
+ * @param session Android Auto Session.
+ * @param mapboxCarMap instance that can be also be found from the [install] function.
+ */
+
+@MapboxExperimental
+class MapboxCarMapSessionInstaller(
+  private val session: Session,
+  private val mapboxCarMap: MapboxCarMap = MapboxCarMap()
+) {
+  private val created = mutableSetOf<MapboxCarMapObserver>()
+  private val started = mutableSetOf<MapboxCarMapObserver>()
+  private val resumed = mutableSetOf<MapboxCarMapObserver>()
+
+  /**
+   * @param observers observers that will be attached while the [Session] is in the
+   * [Lifecycle.State.CREATED] state.
+   */
+  fun created(vararg observers: MapboxCarMapObserver) = apply {
```

Comment:
nit: shall we use `onCreated`?

Reply:
```suggestion
  fun onCreated(vararg observers: MapboxCarMapObserver) = apply {
```

Reply:
I was thinking about this, decided to not include `on*` because `onCreated` typically happens when the event happens. These functions more aligned with the builder pattern.

But naming the lists `onCreated` works too

``` kotlin
    val mapboxCarMap = MapboxCarMapSessionInstaller(session)
      .onCreated(..)
      .onStarted(..)
      .onResumed(..)
      .install()
```